src/components/Network/index.js

```diff
@@ -31,17 +31,27 @@ class Network extends PureComponent {
     this.state = {
       addingChildArrow: null,
       contextMenuItems: [],
-      movingNodePlaceholder: null,
+      movingNode: null,
       nodeToAddChildTo: null,
       newNode: null,
+      svgRef: null,
     };
 
     this.rectRefs = {};
-    this.movingNode = null;
     this.canChangeNodePostion = typeof changeNodePosition === 'function';
     this.onMouseMoveProps = typeof onMouseMove === 'function' ? onMouseMove : noop;
   }
 
+  get canChangeNodePosition() {
+    const { changeNodePosition } = this.props;
+
+    return isFunction(changeNodePosition);
+  }
+
+  handleRef = (svgRef) => {
+    this.setState({ svgRef });
```

Comment:
Avoid adding refs to the state, you can set it directly to the instance and it'll save some checks during an update 😄